### DISCUSSION TITLE
SEO Tags

### DISCUSSION THREAD

**Discussion by SajjadReactDeveloper** [vercel/next.js#52868] [Help]
### Summary

I am adding SEO (meta) tags for social media sharing but facebook and twitter are not picking images and title. Also meta tags are displaying in element but not showing in page source.

**Comment by icyJoseph** [comment]
Most likely running into either:

- an error during SSR which on the client fallbacks to CSR - could be for example because one does `window` or `DOM` access without checking if we are client side first
- a loading flag higher up on the tree that's preventing the meta tags from being rendered server side

**Comment by EdwardTownsend1** [comment]
Before diving deep into more complex solutions, have you tried using Facebook's Debugger tool and Twitter's Card Validator to refresh the social cache?

**Comment by TreyHiggins** [comment]
Ah, the good old conundrum of social media not picking up the right meta tags. I've been down that road and it can be frustrating. Good that your tags are displaying in the element, but not showing up in the source code suggests that they might be added dynamically, which social media crawlers might not pick up effectively.
Since you're working on SEO, have you also looked into [CTR manipulation](https://www.searchseo.io/ctr-manipulation) techniques? This guide explains how click-through rate can be manipulated to improve your rankings. A better understanding of CTR could definitely benefit your overall SEO strategy.
